galley/pkg/server/server.go

```diff
@@ -261,9 +259,7 @@ func RunServer(sa *Args, printf, fatalf shared.FormatFn, livenessProbeController
 		serverReadinessProbe.RegisterProbe(readinessProbeController, "serverReadiness")
 		defer serverReadinessProbe.SetAvailable(errors.New("stopped"))
 	}
-	err = s.Wait()
-	if err != nil {
-		fatalf("Galley Server unexpectedly terminated: %v", err)
-	}
+
+	s.serveWG.Wait()
```

Comment:
Great comments. We should use Public method instead of internal one. I agree with your idea that `RunServer` may not belong here. Maybe have a separate file for it. but how to name it...